Hardware/RAM/GenericRAM.cs

```diff
@@ -1,35 +1,42 @@
-﻿/*
- 
-  This Source Code Form is subject to the terms of the Mozilla Public
-  License, v. 2.0. If a copy of the MPL was not distributed with this
-  file, You can obtain one at http://mozilla.org/MPL/2.0/.
- 
-  Copyright (C) 2012 Michael Möller <mmoeller@openhardwaremonitor.org>
-	
-*/
+﻿// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+// Copyright (C) 2011-2012 Michael Möller <mmoeller@openhardwaremonitor.org>
+// Copyright (C) 2016-2019 Sebastian Grams <https://github.com/sebastian-dev>
+// Copyright (C) 2016-2019 Aqua Computer <https://github.com/aquacomputer, info@aqua-computer.de>
 
 using System.Runtime.InteropServices;
 
 namespace OpenHardwareMonitor.Hardware.RAM {
   internal class GenericRAM : Hardware {
 
-    private Sensor loadSensor;
-    private Sensor usedMemory;
-    private Sensor availableMemory;
+    private Sensor physicalMemoryUsed { get; set; } = null;

```

Comment:
These can be fields.

Reply:
with get/set you see in visual studio all cross references and usage of this objects.
The usage of get/set make no difference in performance.

Reply:
These should be fields, properties are not for that purpose. Unused fields should be highlighted by VS (at least it does here).

Reply:
You see when you use get/set all references to all used code parts, the same like methods. 
the auto generated property like this: private Sensor physicalMemoryUsed { get; set; } = null;
is optimized by jit.
But is better readable. Especially when the project gets bigger.

Reply:
I disagree on the readability part. Properties shouldn't be used for that (and rarely used for that), it's confusing. Properties should be used as if they are public data members, which is not the case for these private implementation details.

Reply:
![code_preview](https://user-images.githubusercontent.com/7898747/61290063-3f143780-a7cb-11e9-92e9-94b25190ebdb.png)

Reply:
I know about CodeLens but that doesn't change anything regarding the readability and purpose of fields versus properties.  You can always right click and select "Find All References" for fields.